### PR TITLE
Add devcontainer configuration

### PR DESCRIPTION
**Pull request by fjakobs**
Allows working with VSCode remote containers and Github CodeSpaces.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.231.3/containers/python-3
+{
+	"name": "Python 3",
+	"build": {
+		"dockerfile": "Dockerfile",
+		"context": "..",
+		"args": { 
+			// Update 'VARIANT' to pick a Python version: 3, 3.10, 3.9, 3.8, 3.7, 3.6
+			// Append -bullseye or -buster to pin to an OS version.
+			// Use -bullseye variants on local on arm64/Apple Silicon.
```
Comment (lennartkats-db) [inline .devcontainer/devcontainer.json:11]: How does that work exactly for non-ARM?

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# See here for image contents: https://github.com/microsoft/vscode-dev-containers/tree/v0.231.3/containers/python-3/.devcontainer/base.Dockerfile
```
Comment (lennartkats-db) [inline .devcontainer/Dockerfile:1]: Could you extend the README.md with a small paragraph about how this is supported for development?

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (fjakobs) [reply .devcontainer/devcontainer.json:11]: This is fully transparent. The ARM enabled docker images are multi-platform and docker will automatically pull the correct platform image. Some of the older images are x86 only. Those work on M1 Macs but only in slow emulation mode.

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (fjakobs) [reply .devcontainer/Dockerfile:1]: Absolutely. Will add it.

Comment (fjakobs) [reply .devcontainer/Dockerfile:1]: done

**Review by nfx — APPROVED** [review]
LGTM

**Pull request merged**